flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/JdbcRowDataLookupFunctionTest.java

```diff
@@ -82,6 +103,114 @@ void testLookup(boolean withFailure) throws Exception {
         assertThat(result).isEqualTo(expected);
     }
 
+    @ParameterizedTest
+    @MethodSource("lookupWithPredicatesProvider")
+    void testEval(TestSpec testSpec) throws Exception {
+        JdbcRowDataLookupFunction lookupFunction =
+                buildRowDataLookupFunctionWithPredicates(
+                        testSpec.withFailure, testSpec.resolvedPredicates, testSpec.pushdownParams);
+
+        ListOutputCollector collector = new ListOutputCollector();
+        lookupFunction.setCollector(collector);
+        lookupFunction.open(null);
+        lookupFunction.eval(testSpec.keys);
+
+        if (testSpec.withFailure) {
+            // Close connection here, and this will be recovered by retry
+            if (lookupFunction.getDbConnection() != null) {
+                lookupFunction.getDbConnection().close();
+            }
+        }
+
+        List<String> result =
+                new ArrayList<>(collector.getOutputs())
+                        .stream().map(RowData::toString).sorted().collect(Collectors.toList());
+        Collections.sort(testSpec.expected);
+        assertThat(result).isEqualTo(testSpec.expected);
+    }
+
+    private static class TestSpec {
```

Comment:
This class should have `toString()`
otherwise it is impossible to navigate between different tests involving this as a parameter

Reply:
good spot - I will add one